Automattic/wp-calypso
# Signup: Update DesignTypeStep and DesignTypeWithStoreStep to use TileGrid

This PR updates `DesignTypeStep` and `DesignTypeWithStoreStep` to use the `TileGrid` component family for displaying all the tiles. We already have the components at hand, so this allows us to reuse lots of the already available styles and get rid of all the duplicated styles we were using. 

This PR is a follow-up to #20639, but addresses the remaining design type components.

My intention is to follow up with so several more cleanup PRs after merging this one, in order to consolidate some styles that are spread out and duplicated among these design type components.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/start/pressable/
* Verify there are no visual/functional differences when comparing to production. Ideally, test on various device sizes and viewports.
* Verify the step works like it did before.
* Go to http://calypso.localhost:3000/start/surveystep/?flags=-signup/atomic-store-flow
* Navigate to the design type step (should be step 2)
* Verify there are no visual/functional differences when comparing to production. Ideally, test on various device sizes and viewports.
* Verify the step works like it did before.

Note: there might be very minor differences in the paddings/margins and copy positioning, but nothing too obvious (unless you compare 1:1), and I'd say that this is a great time to embrace consistency.
